File: Leetcode1886. Determine whether matrix can be obtained through rotation/Solution.ts
function findRotation(mat: number[][], target: number[][]): boolean {
    const n = mat.length;
    // at most 4 rotations
    for (let k = 0; k < 4; k++) {
        // rotation operation
        for (let i = 0; i < Math.floor(n / 2); i++) {
            for (let j = 0; j < Math.floor((n + 1) / 2); j++) {
                [
                    mat[i][j],
                    mat[n - 1 - j][i],
                    mat[n - 1 - i][n - 1 - j],
                    mat[j][n - 1 - i],
                ] = [
                    mat[n - 1 - j][i],
                    mat[n - 1 - i][n - 1 - j],
                    mat[j][n - 1 - i],
                    mat[i][j],
                ];
            }
        }

        if (isEqual(mat, target)) {
            return true;
        }
    }
    return false;
}

function isEqual(mat: number[][], target: number[][]): boolean {
    const n = mat.length;
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            if (mat[i][j] !== target[i][j]) {
                return false;
            }
        }
    }
    return true;
}

File: Leetcode1886. Determine whether matrix can be obtained through rotation/Solution.c
bool isEqual(int** mat, int** target, int n) {
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if (mat[i][j] != target[i][j]) {
                return false;
            }
        }
    }
    return true;
}

bool findRotation(int** mat, int matSize, int* matColSize, int** target,
                  int targetSize, int* targetColSize) {
    int n = matSize;

    // at most 4 rotations
    for (int k = 0; k < 4; k++) {
        // rotation operation
        for (int i = 0; i < n / 2; i++) {
            for (int j = 0; j < (n + 1) / 2; j++) {
                int temp = mat[i][j];
                mat[i][j] = mat[n - 1 - j][i];
                mat[n - 1 - j][i] = mat[n - 1 - i][n - 1 - j];
                mat[n - 1 - i][n - 1 - j] = mat[j][n - 1 - i];
                mat[j][n - 1 - i] = temp;
            }
        }

        if (isEqual(mat, target, n)) {
            return true;
        }
    }
    return false;
}
